### PR TITLE
WIP: Fix traefik routing

### PR DESCRIPTION
**Pull request by ableuler**
- Remove hardcoded `/gitlab` and `/jupyterhub` path service prefixes.
- Remove all defaults for Gitlab and Jupyterhub origins on application and chart level. Both
must be set explicitly.

Note that traefik strips the path from the server url of its service definitions. The service
prefix has to be added through a path AddPrefix middleware. Therefore, we need origin and
servicePrefix as separate chart values.

---
**Do not merge** before corresponding changes have been made to the main Renku chart.

### CODE REVIEW

**Comment by rokroskar** [comment]
If you don't want a PR to be merged, make it a draft pr.

File: helm-chart/renku-gateway/templates/configmap.yaml [hunk 1]
```diff
@@ -79,10 +79,10 @@ data:
       [http.middlewares.development.headers]
         isDevelopment = true
       [http.middlewares.gitlab.AddPrefix]
-        prefix = "/gitlab/api/v4"
+        prefix = "{{ .Values.global.gitlab.servicePrefix }}/api/v4"
       [http.middlewares.jupyterhub.ReplacePathRegex]
         regex = "^/jupyterhub/(.*)"
-        replacement = "/jupyterhub/hub/api/$1"
+        replacement = "{{ .Values.global.jupyterhub.servicePrefix }}/api/$1"
```
Comment (rokroskar) [inline helm-chart/renku-gateway/templates/configmap.yaml]: I think `/hub` is there by default

Comment (ableuler) [reply helm-chart/renku-gateway/templates/configmap.yaml]: > If you don't want a PR to be merged, make it a draft pr.

I would have, but you can't go back once you've pressed the wrong button...

**Comment by ableuler** [comment]
> If you don't want a PR to be merged, make it a draft pr.
I would have, but once you've pressed the wrong button there's no going back...

**Comment by rokroskar** [comment]
I know, that was meant for future reference 😁

The wip in the title also works

**Comment by ableuler** [comment]
outdated, closing!

**Pull request closed without merging**